controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -764,8 +764,6 @@ func TestSecretUpdateOwnerRefAndEnvironmentLabelOnStartup(t *testing.T) {
 	secret = getHostSecret(t, r, host)
 	assert.Equal(t, host.Name, secret.OwnerReferences[0].Name)
 	assert.Equal(t, "BareMetalHost", secret.OwnerReferences[0].Kind)
-	assert.True(t, *secret.OwnerReferences[0].Controller)
-	assert.True(t, *secret.OwnerReferences[0].BlockOwnerDeletion)
```

Comment:
Does this value change? I didn't expect it to.

Reply:
Sorry for late response, I checked the code again and found the `OwnerReference` does not have either of `Controller` and `BlockOwnerDeletion`.

You can see `SetOwnerReference()` in https://github.com/metal3-io/baremetal-operator/blob/main/pkg/secretutils/secret_manager.go#L100 did not set the fields you mentioned (https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L111)

I'm not sure this is expected or not. Any side effect you can see?

Reply:
Yep, these are pointers, so should be checked for `nil` rather than `false`.